### PR TITLE
Update AKS ASO chart

### PR DESCRIPTION
**Pull request by dtzar**
- Support workload identity
- Fix clusterclass API errors
- Improve readme

### CODE REVIEW

**Comment by dtzar** [comment]
@nojnhuh PodIdentityMode is deprecated and the `AUTH_MODE` variable [defaults to workload identity](https://github.com/Azure/azure-service-operator/blob/03d1307def31c7758087ea53808e555bc5854b64/v2/internal/identity/credential_provider.go#L344).  `AUTH_MODE` is also no longer in the public documentation.

File: charts/azure-aks-aso/templates/credentials.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ stringData:
   {{- if .Values.clientSecret }}
   AZURE_CLIENT_SECRET: {{ .Values.clientSecret | quote }}
   {{- end }}
-  {{- if .Values.authMode }}
-  AUTH_MODE: {{ .Values.authMode | quote }}
+  {{- if .Values.useWorkloadIdentity }}
+  USE_WORKLOAD_IDENTITY_AUTH: {{ .Values.useWorkloadIdentity | quote }}
```
Comment (nojnhuh) [inline charts/azure-aks-aso/templates/credentials.yaml]: This was correct before. `USE_WORKLOAD_IDENTITY_AUTH` is only valid in the global aso-controller-settings Secret where this is a per-resource Secret which has a slightly different format. The global aso-controller-settings secret defaults to managed identity unless `USE_WORKLOAD_IDENTITY_AUTH: true` is set and per-resource secrets like this default to workload identity unless `AUTHMODE: podidentity` is set.

https://azure.github.io/azure-service-operator/guide/authentication/credential-format/

File: charts/azure-aks-aso/templates/credentials.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ metadata:
     {{- include "capz.commonLabels" . | nindent 4 }}
   annotations:
     helm.sh/resource-policy: keep # credentials have to be available throughout a delete operation
-stringData:
+stringData: # https://azure.github.io/azure-service-operator/guide/aso-controller-settings-options/
```
Comment (nojnhuh) [inline charts/azure-aks-aso/templates/credentials.yaml]: This link describes the global aso-controller-settings Secret but this is a per-resource Secret that only contains credentials and has a different set of known fields, so a better link would be here: https://azure.github.io/azure-service-operator/guide/authentication/credential-format/

File: charts/azure-aks-aso/templates/credentials.yaml [hunk 3]
```diff
@@ -23,4 +23,7 @@ stringData:
   {{- if .Values.authMode }}
   AUTH_MODE: {{ .Values.authMode | quote }}
   {{- end }}
+  {{- if .Values.useWorkloadIdentity }}
+  USE_WORKLOAD_IDENTITY_AUTH: {{ .Values.useWorkloadIdentity | quote }}
+  {{- end }}
```
Comment (nojnhuh) [inline charts/azure-aks-aso/templates/credentials.yaml]: ASO doesn't know how to make use of this `USE_WORKLOAD_IDENTITY_AUTH` variable in this Secret, so it should probably be removed. That field is only read on the global aso-controller-settings Secret, not these per-resource credential Secrets.

**Review by nojnhuh — APPROVED** [review]
lgtm

**Pull request merged**